src/ui/buttons/props.js

```diff
@@ -762,18 +761,22 @@ export function normalizeButtonMessage(
 
   if (typeof offer !== "undefined") {
     if (!Array.isArray(offer)) {
-      throw new TypeError(
-        `Expected message.offer to be an array of strings, got: ${String(
-          offer
-        )}`
-      );
+      if (typeof offer !== "string") {
+        throw new TypeError(
+          `Expected message.offer to be an array of strings, got: ${String(
+            offer
+          )}`
+        );
+      }
+      offer = [offer];
     }
     const invalidOffers = offer.filter(
       (o) => !values(MESSAGE_OFFER).includes(o)
     );
     if (invalidOffers.length > 0) {
       throw new Error(`Invalid offer(s): ${invalidOffers.join(",")}`);
     }
+    offer = offer.join(",");
```

Comment:
Does this change breaks any existing clients of this method?

Reply:
yes, clients that are expecting offer as an array should now expect it as a joined string. They will also have to be updated to receive a string.